codesign/lib/src/codesign_context.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'package:file/local.dart';
+import 'package:process/process.dart';
+import 'package:codesign/codesign.dart';
+import 'package:file/file.dart';
+
+class CodesignContext {
+  CodesignContext({
+    required this.codesignCertName,
+    required this.codesignPrimaryBundleId,
+    required this.codesignUserName,
+    required this.appSpecificPassword,
+    required this.codesignAppstoreId,
+    required this.codesignTeamId,
+    required this.codesignFilepaths,
+    required this.commitHash,
+    this.production = false,
+  });
+
+  final String codesignCertName;
+  final String codesignPrimaryBundleId;
+  final String codesignUserName;
+  final String appSpecificPassword;
+  final String codesignAppstoreId;
+  final String codesignTeamId;
+  final List<String> codesignFilepaths;
+  final String commitHash;
+  final bool production;
+  Directory? tempDir;
```

Comment:
Does this need to be an instance variable? Looks like it can just be a nonnull local set in `run`.

Reply:
Yep l put it as instance variable because later in tests I override `codesignVisitor` with `fakeCodesignVisitor`, and set the tempDir of the fakeCodesignVisitor with `codesignContext.tempDir`.